tests/Test.ADAL.NET.Unit.net45/RegressionTests.cs

```diff
@@ -46,6 +47,52 @@ public void Initialize()
             InstanceDiscovery.InstanceCache.Clear();
         }
 
+        [TestMethod]
+        public async Task Double504Async()
```

Comment:
Sadly this test, as well as repeated attempts to repro this with Fiddler AutoResponder, fail to repro the issue.